File: listings/child-process/ipc/ipc.c
#include <unistd.h>
#include <stdio.h>
#include <sys/socket.h>
#include <string.h>

int main () {
  char buffer[80];
  int rc;

  while (rc = recv(3, buffer, sizeof(buffer), 0)) {
    printf("child received: %s", buffer);
    send(3, buffer, rc, 0);
    memset(buffer, 0, sizeof(buffer));
  }
}

File: listings/child-process/ipc/ipc.js
var cp = require('child_process')

var child = cp.spawn('./ipc', [], { stdio: [ 0, 1, 2, 'ipc'] })
child.on('message', function (msg) {
  console.log('parent received:', msg)
})

child.send('monkeys')
child.send(true)
child.send('larger than the characters you have in your 80 char buffer object, hope you survive')

setTimeout(function () {
  child.send(23)
  child.send({ an: 'object' })
}, 2000)

console.log(child)
